doc/scripts/genrest.py

```diff
@@ -135,6 +143,22 @@ def init():
 
         modules.append((title, suffix, abspath, desc_path))
 
+    no_index_modules = []
+    for module_spec in args.no_index_modules:
+        spec_parts = module_spec.split(",")
```

Comment:
```suggestion
        spec_parts = module_spec.split(",", 1)
```

This is a bit more robust for paths with commas.

```
>>> foo = 'title,path-with-commas,in-it'
>>> foo.split(',')
['title', 'path-with-commas', 'in-it']
>>> foo.split(',', 1)
['title', 'path-with-commas,in-it']
```

Perhaps something to apply to the other argument this one is modeled after.

Reply:
Done.

Reply:
> Perhaps something to apply to the other argument this one is modeled after.

What about this? I had assumed this patch was just copy/pasting the approach taken by the other argument, but expanding the diff now I see that they work differently. The other argument allows commas by passing a doubled comma, `,,`.

Why is this one different?

Reply:
> Why is this one different?

I don't see why I'd change the other if it's currently working, and I haven't even seen any examples outside of nRF Connect SDK, so I don't know how people are using it. For my parameter I just need two entries because it does not create new files, it simply resolves paths to a `title` in the generated docs. Or perhaps I didn't understand exactly what you're asking for...

Reply:
The commit log says:

```
This add a new option `--no-index-modules` which works similarly to
`--modules` but does not generated index pages, only retains the
tweaking of how paths are displayed on symbol information pages,
showing '<title>/path/within/module/Kconfig' for paths that fall
within modules.
```

but --no-index-modules doesn't work very similarly to --modules in terms of how its users can include paths with literal commas, which I find confusing and inconsistent for no reason I could understand. Is there a reason? That is what I am asking when I say "why is this one different?"

Reply:
I honestly don't understand, I updated as suggested previously, and paths with commas do work:

```python
>>> 'SomeTitle,path,with,commas'.split(',', 1)
['SomeTitle', 'path,with,commas']
```
Perhaps you could give me an example of what you have in mind that would not work?

Reply:
Did you see this update from me?

```
 I had assumed this patch was just copy/pasting the approach taken by the other argument, but expanding the diff now I see that they work differently. The other argument allows commas by passing a doubled comma, ,,.
```

Reply:
That is, I had:

1. assumed --no-index-modules was copy/pasting the same approach to commas taken by --modules
2. noticed --no-index-modules didn't allow paths with commas, suggested an alternative that would, asked if the two could be kept consistent
3. on second review, noticed that --no-index-modules in fact did not copy/paste the approach taken by --modules with regards to commas
4. asked why they were inconsistent
5. am still asking why they were not consistent in the first place (is there a good reason?), and if there's no reason, asking for them to be treated consistently so users will know how to use both of them

Reply:
I updated it, so now it should be following the same `,,` => `,` rules.